### PR TITLE
[WIP] Objc modular imports

### PR DESCRIPTION
**Pull request by ob**
This adds support for using modular imports (`@import module`)  from `objc_library` rules. 

Implementation note:

I again had to remove the extraneous `CcCompilationContext` mentioned in #5954, and still have the same question from that PR.

I also cleaned up what looked like unused code (that's why there are two commits).

### CODE REVIEW

File: src/test/java/com/google/devtools/build/lib/rules/objc/ObjcLibraryTest.java [hunk 1]
```diff
@@ -380,7 +380,7 @@ public void testProvidesObjcHeadersWithDotMFiles() throws Exception {
     assertThat(getArifactPaths(target, HEADER))
         .containsExactly("objc/a.h", "objc/b.h", "objc/f.m");
     assertThat(getArifactPaths(depender, HEADER))
-        .containsExactly("objc/a.h", "objc/b.h", "objc/f.m", "objc2/d.h", "objc2/e.m");
+        .containsExactly("objc/a.h", "objc/b.h", "objc/f.m", "objc/private.h", "objc2/d.h", "objc2/e.m");
```
Comment (allevato) [inline src/test/java/com/google/devtools/build/lib/rules/objc/ObjcLibraryTest.java]: This is checking the `header` field of the `ObjcProvider`, right? AFAIK, it definitely should _not_ be propagating private headers that are only listed in `srcs`.

Comment (ob) [reply src/test/java/com/google/devtools/build/lib/rules/objc/ObjcLibraryTest.java]: I was going by [this comment](https://github.com/bazelbuild/bazel/pull/5954#issuecomment-417259877) from #5954.

**Comment by ob** [comment]
One thing to consider is that with too many dependencies this has the potential to exceed the maximum command line argument. I should probably pass all the flags as a response file.

**Comment by ob** [comment]
I am a bit confused why [here](https://github.com/bazelbuild/bazel/blob/aaba6feff7606f72dd4a2c14e837f56761628d33/src/main/java/com/google/devtools/build/lib/rules/objc/CompilationSupport.java#L1247) `-fmodules` is only passed to the compiler when Bazel itself generates the module map and *not* when the user provided a module map. I want to pass `-fmodules` in both cases.

**Comment by ob** [comment]
This is no longer being pursued in this form, I'll take another crack once the C++ rules are in StarLark.

**Pull request closed without merging**